### DISCUSSION TITLE
Adding link to "edit or improve this page" to lyric template

### DISCUSSION THREAD

**Discussion by andy5995** [Lyrics/lyrics#491] [Ideas]
@snshn I'm sure you've seen this link on [some websites that are hosted on GitHub](https://mesonbuild.com/Style-guide.html). What do you think about adding a link to "edit or improve this page" to the template that displays the lyrics for an individual song?

**Comment by snshn** [comment]
I like this idea; the only thing that worries me is distracting the user from the content... but one link somewhere under the lyrics text, among/below the metadata shouldn't slow down the render or navigation speed on the site (no extra CSS needed, just HTML).

Perhaps it could be a boolean config variable for build.py in lyrics-website-generator.git for people who don't want it to be enabled on their lyrics websites.

Pro: bring more contributors and traffic to this github repo
Con: one small step closer to other lyrics websites that are hard to read and full of ads

@defanor what do you think?

**Comment by defanor** [comment]
I agree about the listed pros and cons. It's hard to estimate how many
additional useful contributions this would bring: ways to contribute
seem to be easily discoverable as it is, and I guess that the other
barriers (potential copyright law violations, need to have/use a GitHub
account or to write an email message) may play a bigger role, though
that's also hard to estimate.

I'd suggest that we try it and see how it works (for both contribution
rate and usability), but the contributions so far are rather sparse, so
it'd be hard to see slight changes in their rate (assuming that the
difference won't be huge), and usability changes would be even harder to
detect. Maybe this experiment should be left for later, when (hopefully)
there will be more visitors.

**Comment by snshn** [comment]
The copyright thing should be fine since there's no money made with this project (no ads, no contributions), and materials are provided for non-commercial educational use, so it's as fair use as it can be. And DMCA is mentioned on the website's homepage and in the README.md, so even if this project gets an official request to take some materials down, we can do that with no problem, backing it up with a test to blacklist those artists from the database to exclude possibility of future additions. Even if the repo gets shut down by GitHub the way youtube-dl got nuked, somebody else will continue the work somewhere else, likely on dark web. Many Internet projects are likely to be much more decentralized and self-hosted, this one is just one of them at its early stage. We've done what we could to protect it from being removed for any reason without warning, hence I wouldn't worry too much about any of that until it happens.

I agree that it can be viewed as a nice experiment, but disagree with implementing it later being a better idea. I'm thinking that due to current low traffic to the repo we could easier see if the "edit this file" button does anything, it should even show more traffic in the Insights of this repo on GitHub, indicating if more people are coming from the GitHub Pages site via those links.

**Comment by defanor** [comment]
As mentioned over XMPP, traffic may indicate just crawler traffic, while
the goal is to increase contribution rate. Though we could try it now,
and even if there will be no measurable increase in contributions, try
it again later: should be easy enough if it'll be a config variable, as
you've suggested above.

**Comment by snshn** [comment]
It is done, can be seen on the website.
